sched/queue/queue.go

```diff
@@ -12,6 +12,11 @@ type Queue interface {
 	// Enqueue enqueues a job and assigns it an ID. Errors may be
 	// an *InvalidJobRequest or a *CanNotScheduleNow
 	Enqueue(job sched.Job) (string, error)
+
+	Chan() chan WorkItem
+
+	Status(jobId string) (sched.JobStatus, error)
```

Comment:
I think this introduces added complexity and consistency issues if we have to write to a durable queue and an in memory one.  The only reason to do this is if we have latency requirements aka use it as a cache.  I would prefer to just read off the durable log and not have both, less code and less potential bugs.  

I was under the impression that the in memory queue is just for testing